src/main/java/io/jenkins/update_center/ArtifactoryRepositoryImpl.java

```diff
@@ -79,15 +79,20 @@ protected Set<ArtifactCoordinates> listAllJenkinsWars(String groupId) throws IOE
     }
 
     private static boolean containsIllegalChars(String test) {
-        return !test.chars().allMatch(c -> c >= 0x2B && c < 0x7B);
+        return !test.chars().allMatch(c ->
+                c >= 0x30 && c <= 0x39 // allow digits 0-9
+                || c >= 0x41 && c <= 0x5A // allows letter A-Z
+                || c >= 0x61 && c <= 0x7A // allows letter a-z
+                || c == 0x2B || c >= 0x2D && c <= 0x2F || c == 0x5F // allows: +-./_
```

Comment:
Looks good, but why do you use hex codes?
```suggestion
        return !test.chars().allMatch(c ->
                c >= '0' && c <= '9'
                || c >= 'A' && c <= 'Z'
                || c >= 'a' && c <= 'z'
                || c == '+' || c == '-' || c <= '.' || c == '/' || c == '_'
```

Reply:
This went through a few iterations, and started by having range boundaries that made no sense as characters (e.g. `+` through `9` or `_` through `z`), and I didn't adapt the representation when I went with more easily consumable ranges.